substratevm/src/com.oracle.svm.test.jdk11/src/com/oracle/svm/test/jdk11/jfr/utils/JFRFileParser.java

```diff
@@ -0,0 +1,168 @@
+import com.oracle.svm.test.jdk11.jfr.utils.pools.SymbolConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.pools.ThreadConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.pools.ThreadGroupConstantPoolParser;
+import com.oracle.svm.test.jdk11.jfr.utils.pools.ThreadStateConstantPoolParser;
+import jdk.jfr.Recording;
+
+public class JFRFileParser {
+
+    private static class Positions {
+
+        private final long constantPoolPosition;
+        private final long metadataPosition;
+
+        public Positions(long constantPoolPosition, long metadataPositions) {
+            this.constantPoolPosition = constantPoolPosition;
+            this.metadataPosition = metadataPositions;
+        }
+
+        public long getConstantPoolPosition() {
+            return constantPoolPosition;
+        }
+
+        public long getMetadataPosition() {
+            return metadataPosition;
+        }
+    }
+
+    private static final NotFoundConstantPoolParser notFoundConstantPoolParser;
+    private static final HashMap<Long, ConstantPoolParser> supportedConstantPools;
+
+    static {
+        supportedConstantPools = new HashMap<>();
+        notFoundConstantPoolParser = new NotFoundConstantPoolParser();
+
+        supportedConstantPools.put(JfrTypes.Class.getId(), new ClassConstantPoolParser());
+        supportedConstantPools.put(JfrTypes.ClassLoader.getId(), new ClassLoaderConstantPoolParser());
+        supportedConstantPools.put(JfrTypes.Package.getId(), new PackageConstantPoolParser());
+        supportedConstantPools.put(JfrTypes.Module.getId(), new ModuleConstantPoolParser());
+
+        supportedConstantPools.put(JfrTypes.Symbol.getId(), new SymbolConstantPoolParser());
+        supportedConstantPools.put(JfrTypes.Method.getId(), new MethodConstantPoolParser());
+        supportedConstantPools.put(JfrTypes.StackTrace.getId(), new StacktraceConstantPoolParser());
+        supportedConstantPools.put(JfrTypes.FrameType.getId(), new FrameTypeConstantPoolParser());
+
+        supportedConstantPools.put(JfrTypes.Thread.getId(), new ThreadConstantPoolParser());
+        supportedConstantPools.put(JfrTypes.ThreadGroup.getId(), new ThreadGroupConstantPoolParser());
+        supportedConstantPools.put(JfrTypes.ThreadState.getId(), new ThreadStateConstantPoolParser());
+    }
+
+    private static Positions parserFileHeader(RecordingInput input) throws IOException {
+        byte[] fileMagic = new byte[JfrChunkWriter.FILE_MAGIC.length];
+        input.readFully(fileMagic); // File magic.
+        assertEquals("File magic is not correct!", new String(JfrChunkWriter.FILE_MAGIC), new String(fileMagic));
+        assertEquals("JFR version major is not correct!", JfrChunkWriter.JFR_VERSION_MAJOR, input.readRawShort());
+        assertEquals("JFR version minor is not correct!", JfrChunkWriter.JFR_VERSION_MINOR, input.readRawShort());
+        assertTrue("Chunk size is null!", input.readRawLong() > 0); // Chunk size.
```

Comment:
The error message mentions `null` but the check is `> 0`, which is confusing because I'd expect `!= null` based on the message. Maybe s/null/invalid ?

Reply:
This applies to some more of the asserts below too.

Reply:
I have replaced `null` with `invalid`.